core/Fetch.cpp

```diff
@@ -63,14 +63,16 @@ namespace olympia
         for(uint32_t i = 0; i < upper; ++i)
         {
             InstPtr ex_inst = inst_generator_->getNextInst(my_clk_);
-            if(SPARTA_EXPECT_TRUE(nullptr != ex_inst))
-            {
+            if(SPARTA_EXPECT_TRUE(nullptr != ex_inst)){
                 ex_inst->setSpeculative(speculative_path_);
                 insts_to_send->emplace_back(ex_inst);
 
                 ILOG("Sending: " << ex_inst << " down the pipe");
+                // we break after vset to break up instruction group
+                if(ex_inst->isVset()) { break; }
```

Comment:
Why do we need this? Looks like you have stalls working in Decode so can this be removed?

Reply:
Removed in [73ca9f4](https://github.com/riscv-software-src/riscv-perf-model/pull/167/commits/73ca9f4adb0bce0b8756a9b1ab3d88bf96c589bd)